project-koku/koku
# [COST-4120] Fix capacity instance counts.

## Jira Ticket

[COST-4120](https://issues.redhat.com/browse/COST-4120)

## Description

This change will change the capacity instance counts for OCP. 

Previously we calculated in the same manner as what we did for capacity, where we summed the values over time. However instead we are just finding the max value. There is a big change to the OCP cluster instance counts:

_if you have 5 nodes on a cluster, we should be summing the number of cores on those nodes. So if a cluster has 5 nodes, and each node has 10 CPU, i would expect to see 50 CPU there_

Since I needed to create the mapping I needed to do a different query that allowed me to group by node. Therefore, I move the cluster annotations to their place in the provider map. 


## Testing

1. Checkout Branch
2. load-test-customer-data
3. Check over all capacity:

With these recent changes the total capacity between cluster & node for the same time range should be the same:

*Memory*
1. Cluster
```
http://localhost:8000/api/cost-management/v1/reports/openshift/memory/
```
Example:
```
 "capacity": {
                "value": 14016.0,
                "units": "GB-Hours",
                "unused": 8130.85,
                "unused_percent": 58.011201484018265,
                "count": 584.0,
                "count_units": "GB"
            },
```
1. Node
```
http://localhost:8000/api/cost-management/v1/reports/openshift/memory/?group_by[node]=*
```
Example:
```
"capacity": {
                "value": 14016.0,
                "units": "GB-Hours",
                "unused": 8130.85,
                "unused_percent": 58.011201484018265,
                "count": 584.0,
                "count_units": "GB"
            },
```

## Notes

...
